devices/Ds1302/Ds1302.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Device.Model;
+using Iot.Device.Common;
+
+namespace Iot.Device.Rtc
+{
+    /// <summary>
+    /// Realtime Clock Ds1302.
+    /// </summary>
+    [Interface("Realtime Clock Ds1302")]
+    public class Ds1302 : RtcBase
+    {
+        private readonly GpioPin _clockPin;
+        private readonly GpioPin _resetPin;
+        private readonly GpioPin _dataPin;
+        private readonly byte enableWrite = 0b1000_0000;
+        private readonly byte disableWrite = 0b0000_0000;
+        private readonly byte enableWriteOrRead = 0b1000_0001;
+        private readonly bool _shouldDispose;
+        private PinMode _currentDataPinDirection;
+        private GpioController _controller;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Ds1302"/> class.
+        /// </summary>
+        /// <param name="clockPin">Clock pin number.</param>
+        /// <param name="dataPin">Data pin number.</param>
+        /// <param name="resetPin">Reset pin number.</param>
+        /// <param name="controller">Gpio controller.</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller.</param>
+        /// <exception cref="ArgumentNullException">When controller was not provided.</exception>
+        public Ds1302(int clockPin, int dataPin, int resetPin, GpioController? controller = null, bool shouldDispose = true)
+        {
+            _controller = controller ?? new GpioController();
+            _shouldDispose = shouldDispose || controller is null;
+
+            if (clockPin < 0 || dataPin < 0 || resetPin < 0)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            _clockPin = _controller.OpenPin(clockPin, PinMode.Output);
+            _resetPin = _controller.OpenPin(resetPin, PinMode.Output);
+            _dataPin = _controller.OpenPin(dataPin, PinMode.Input);
+
+            _resetPin.Write(PinValue.Low);
```

Comment:
and then lower this if the pin is positive only.

so all up, it would look like this:

```csharp
if (clockPin < 0 || dataPin < 0)
{
        throw new ArgumentOutOfRangeException();
 }

 _clockPin = _controller.OpenPin(clockPin, PinMode.Output);
_dataPin = _controller.OpenPin(dataPin, PinMode.Input);

if (resetPin >= 0)
{
    _resetPin = _controller.OpenPin(resetPin, PinMode.Output);
    _resetPin.Write(PinValue.Low);
}
```

Reply:
I beg to differ here, I have no knowledge about eletronics but from logical perspective to make communication successful you have to set the resetPin to be high at some point -> see method PrepareRead. So would logical to have it open first right? The way you are proposing you won't open the pin at all if it was negative (shorted to ground) so in my opinion it suppose to fail on write as pin was not open or simply do not work as intented? 

Having that in mind I suggest to keep it always positive.

Reply:
OK. So I checked the datasheet and understand now. The "Reset" name is not appropriate. In old datasheet it was called like this but it's not a reset. Reason why the in name changed for "CE" which should refer to "CommunicationEnabled". So I would suggest then to adjust either the name of the pin, either add a comment about this.

![image](https://user-images.githubusercontent.com/8145578/187170962-a4dca2b8-9f5b-4f70-9d02-f260785bdcbe.png)